bftengine/src/bftengine/ControllerWithSimpleHistory.hpp

```diff
@@ -66,6 +73,26 @@ class ControllerWithSimpleHistory : public ControllerBase {
   // Returns true in case a sequence number is within the range of the current window.
   bool insideActiveWindow(const SeqNum& n);
 
+  // Returns lower or upper bound, in case val is out of range, otherwise returns val.
+  template <typename T>
+  static const T& normalizeToRange(const T& lower, const T& upper, const T& val) {
```

Comment:
Actually, I expected these 3 functions to be free(not a part of a particular class).
Leaving them as a part of the class makes its interface less clean.
Honestly, I believe that they are a good candidate for being an independent utility:)

Reply:
I think it only makes sense to extract them as an independent utility only if another component needs to use them. Which would those components be? I think a static method is also an acceptable model for representing "pure" functions (no side effects).

Reply:
I tend to leave it as is, since utils tend to become a mess of unused functions, and beside this class I don't see any usage.  I agree with @f-squirrel that it could go to a util lib as well.
Anyway its such a minor functionality that it's not worth the effort.